packages/core/js-client/src/fetchers/node.ts

```diff
@@ -15,22 +15,37 @@
  */
 
 import fs from 'fs';
-import url from 'url';
 import path from 'path';
+import module from 'module';
 
-export async function fetchResource(assetPath: string, version: string) {
+export async function fetchResource(pkg: string, assetPath: string) {
+    const require = module.createRequire(import.meta.url);
+    const packagePathIndex = require.resolve(pkg);
+    
+    // Ensure that windows path is converted to posix path. So we can find a package 
+    const posixPath = packagePathIndex.split(path.sep).join(path.posix.sep);
+    
+    const matches = new RegExp(`(.+${pkg})`).exec(posixPath);
+    
+    const packagePath = matches?.[0];
+    
+    if (!packagePath) {
+        throw new Error(`Cannot find dependency ${pkg} in path ${posixPath}`);
+    }
```

Comment:
is it possible to do this or am I missing something?
```suggestion
    const packagePath = packagePathIndex.split(path.sep).join(path.posix.sep);
    const matches = new RegExp(`.+${pkg}`).test(packagePath);    
    if (!matches) {
        throw new Error(`Cannot find dependency ${pkg} in path ${packagePath}`);
    }    
```

Reply:
`exec` gives you capturing value from regex which i use later
`test` gives you boolean value whether string matches or not

Reply:
thanks for the explanation